crates/shared/src/http_solver/model.rs

```diff
@@ -159,7 +159,7 @@ pub struct InteractionData {
     ///
     /// `AMM -> GPv2Settlement`
     pub outputs: Vec<TokenAmount>,
-    pub exec_plan: Option<ExecutionPlan>,
+    pub exec_plan: ExecutionPlan,
```

Comment:
One "code-smell" is that interactions are encoded in a vector which already has a notion of order, and then we need to add an additional ordering coordinates within that vector.

I do think its necessary (so you can interleave AMM executions with raw interactions), but maybe we can make it a bit more ergonomic where for `InteractionData`, we automatically assign:
```js
{
  "exec_plan": {
    "sequence": usize::MAX, // default to coming after all other AMMs, as it does now
    "position": indexOf(interaction), // its position in the interactions vector
    "internal": false, // same as for other interactions AFAIU.
  },
}
```

So, we do some custom de-serialization in order to take advantage of the fact that we already have ordering information in the JSON data and make it more ergonomic to use.

Reply:
@sunce86 - just checking if you saw this comment as this is a nice way to not require execution plans for `InteractionData` in a way that is 100% backwards compatible.

Reply:
I added a special temporary function to add missing exec plans for interactions that are missing one.
(Implementing a custom deserializer was too verbose).

In any case, I am not a fan of this fix since we introduce unnecessary complexity to our system instead of being more strict. By being less strict I feel we just fix one edge case but we open doors to a large number of different cases/potential bugs. On the other hand, I find it ok as a temporary solution as with this, we support most of the external solvers sending `null` as an execution plan (which is happening right now).

But, once again, I would like to end up requesting valid execution plan by solvers.

Reply:
Let me know if the fix is ok so I can add unit tests.

Reply:
I think the fix here is fine, but agree with:

>  I am not a fan of this fix since we introduce unnecessary complexity to our system instead of being more strict

I was suggesting at the `Deserialize` level just so we don't have to deal with `Option<ExecutionPlan>` in code. If this is not really feasible without huge changes, then I also slightly prefer it without that latest fix.